nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/BackingServiceConfigurator.java

```diff
@@ -0,0 +1,265 @@
+        RetryPolicy retryPolicy = new RetryPolicy().withMaxRetries(0);
+
+        Properties userConfig = configurationGenerator.getUserConfig();
+        if ("true".equals((userConfig.getProperty(PARAM_RETRY_POLICY_ENABLED,"false")))) {
+
+            int maxRetries = Integer.parseInt(userConfig.getProperty(PARAM_RETRY_POLICY_MAX_RETRIES, "20"));
+            int delay = Integer.parseInt(userConfig.getProperty(PARAM_RETRY_POLICY_DELAY_IN_MS, "5000"));
+
+            retryPolicy = retryPolicy.retryOn(ConfigurationException.class).withMaxRetries(maxRetries).withDelay(delay,
+                    TimeUnit.MILLISECONDS);
+        }
+        return retryPolicy;
+    }
+
+    protected Collection<BackingChecker> getCheckers() throws ConfigurationException {
+
+        if (checkers == null) {
+            checkers = new HashSet<>();
+
+            for (String template : configurationGenerator.getTemplateList()) {
+                try {
+                    File templateDir = configurationGenerator.getTemplateConf(template).getParentFile();
+                    String classPath = getClasspathForTemplate(template);
+                    String checkClass = configurationGenerator.getUserConfig().getProperty(template + ".check.class");
+
+                    Optional<URLClassLoader> ucl = getClassLoaderForTemplate(templateDir, classPath);
+                    if (ucl.isPresent()) {
+                        Class<?> klass = Class.forName(checkClass, true, ucl.get());
+                        checkers.add((BackingChecker) klass.newInstance());
+                    }
+
+                } catch (IOException e) {
+                    log.warn("Unable to read check configuration for template : " + template, e);
+                } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+                    throw new ConfigurationException("Unable to check configuration for backing service " + template,
+                            e);
+                }
+            }
+            checkers.add(new DBCheck());
+        }
+        return checkers;
+    }
+
+    /**
+     * Read the classpath parameter from the template and expand parameters with their value. It allow classpath of the
+     * form ${nuxeo.home}/nxserver/bundles/... VisibleForTesting
+     *
+     * @param template The name of the template
+     * @return
+     */
+    String getClasspathForTemplate(String template) {
+        String classPath = configurationGenerator.getUserConfig().getProperty(template + ".check.classpath");
+        final TextTemplate templateParser = new TextTemplate(configurationGenerator.getUserConfig());
+        classPath = templateParser.processText(classPath);
+        return classPath;
+    }
+
+    /**
+     * Build a ClassLoader based on the classpath definition of a template.
+     *
+     * @param templateDir
+     * @param classPath
+     * @return
+     * @throws ConfigurationException
+     * @throws IOException
```

Comment:
Remove empty `@param`/`@return`/`@throws` if you don't fill them in this `protected` method.